components/bl-autocomplete-component/src/index.js

```diff
@@ -21,10 +21,20 @@ export default function AutocompleteComponent({ component, eventHandlers }) {
     setOptionsList(validate(options));
   }, [options]);
 
+  const hasGroup = optionsList[0]?.hasOwnProperty("groupLabel");
   const autocompleteHeight = autocompleteRef.current?.getBoundingClientRect()?.height;
-  const filteredOptions = optionsList.filter(({ label }) => (
-    label.toLowerCase().includes(inputValue.toLowerCase())
-  ));
+
+  const filteredOptions = optionsList.reduce((acc, item) => {
+    const isMatched = label => label.toLowerCase().includes(inputValue.toLowerCase());
+
+    if (hasGroup) {
+      const filteredChildren = item.children.filter(child => isMatched(child.label));
+
+      return filteredChildren.length ? [...acc, { ...item, children: filteredChildren }] : acc;
+    }
+
+    return isMatched(item.label) ? [...acc, item] : acc;
+  }, []);
```

Comment:
move to hook or function. Also it could be memoized and maybe we need to add debounce here.